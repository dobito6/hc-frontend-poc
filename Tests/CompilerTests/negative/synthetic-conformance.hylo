// An indestructivble struct.
struct Indestructible {}

// A generic struct.
struct Box<T> { var x: T }

// Conformance is structurally synthesizable.
struct A: Deinitializable {}

// Conformance is structurally synthesizable.
struct B: Deinitializable {
  var x: {x: Void}  // structural conformance
  var y: A          // user-defined conformance
}

// Conformance is synthesizable using the where clause.
given <X where X: Deinitializable> Box<X>: Deinitializable {}

// Conformance is not synthesizable.
given <X> Box<X>: Deinitializable {}

// A struct whose contents isn't structurally deinitializable.
struct NotStructurallyDeinitializable: Deinitializable {
  var x: Indestructible
}
