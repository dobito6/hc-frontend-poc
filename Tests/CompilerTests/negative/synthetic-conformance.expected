negative/synthetic-conformance.hylo:20.1-6: error: no implementation of 'deinit'
given <X> Box<X> is Deinitializable {}
~~~~~
negative/synthetic-conformance.hylo:23.39-41: error: no implementation of 'deinit'
struct NotStructurallyDeinitializable is Deinitializable {
                                      ~~
